### PR TITLE
Vb 1926 update orchestration service to acquire full username and return it to the UI for each entry

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
Get the usernames for a given userid by making a call to the HMPPS Auth service

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+  fun getFullVisitDetailsByReference(
+    reference: String,
+  ): VisitDto? {
+    val visitDto = visitSchedulerClient.getVisitByReference(reference)
+    visitDto?.also {
+      val createdByMono = getUserDetails(visitDto.createdBy)
+      val updatedByMono = getUserDetails(visitDto.updatedBy)
+      val cancelledByMono = getUserDetails(visitDto.cancelledBy)
+
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:28]: wonder if you could do something clever here and if the visitDto.createdBy is the same as visitDto.updatedBy or the visitDto.cancelledBy only call it once?

```
val createdByMono = getUserDetails(visitDto.createdBy)
val updatedByMono = if (visitDto.createdBy==visitDto.updatedBy) createdByMono else getUserDetails(visitDto.updatedBy)
```
not sure how that would work with the Mono zip though!
Just i'm concerned about Auth and it's performance.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/VisitDto.kt [hunk 2]
```diff
@@ -42,12 +42,18 @@ class VisitDto(
...
+  var createdByFullName: String? = null,
+  @Schema(description = "Updated By ID - user id for the user who last updated the visit", example = "AB12345A", required = false)
   val updatedBy: String? = null,
-  @Schema(description = "Cancelled By - user id for the user who cancelled the visit", example = "AB12345A", required = false)
+  @Schema(description = "Updated By Name - full name for the user who last updated the visit", example = "John Smith", required = false)
+  var updatedByFullName: String? = null,
+  @Schema(description = "Cancelled By ID - user id for the user who cancelled the visit", example = "AB12345A", required = false)
   val cancelledBy: String? = null,
+  @Schema(description = "Cancelled By Name - full name for the user who cancelled the visit", example = "John Smith", required = false)
+  var cancelledByFullName: String? = null,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/VisitDto.kt:56]: wonder if these should be in a sub DTO like
```
var createdBy : ActionedByUserDetailsDTO
var updatedBy : ActionedByUserDetailsDTO?=null
var canceledBy : ActionedByUserDetailsDTO?=null
```

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:28]: I will explore moving usernames to a Set and making fewer calls.

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:28]: raised a story to pick these up later - https://dsdmoj.atlassian.net/browse/VB-2045

**Review by AledWEvans — APPROVED** [review]
LGTM, assume other work will happen in the not to distant future

**Pull request merged**